src/content/docs/redact/reference.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: "Arcjet Redact reference"
+description: "Reference for the Arcjet redaction library"
+---
+
+import {
+  Tabs,
+  TabItem,
+  LinkCard,
+  CardGrid,
+  Code,
+} from "@astrojs/starlight/components";
+import WhatIsArcjet from "/src/components/WhatIsArcjet.astro";
+import FAQs from "/src/components/FAQs.astro";
+import CustomEntities from "/src/content/docs/redact/snippets/CustomEntities.ts?raw";
+import CustomRedact from "/src/content/docs/redact/snippets/CustomRedact.ts?raw";
+import QuickStartUnredact from "/src/content/docs/redact/snippets/QuickStartUnredact.ts?raw";
+import Comments from "/src/components/Comments.astro";
+
+The Arcjet Redaction library makes it easy to redact sensitive information
+locally. It is a utility library independent of the main Arcjet SDK so can be
+used with or without other Arcjet rules.
+
+<WhatIsArcjet />
+
+## Configuration
+
+The configuration definition is:
+
+```ts
+type RedactOptions = {
+  entities?: Array<SensitiveInfoType>;
+  contextWindowSize?: number;
+  detect?: (tokens: string[]) -> Array<SensitiveInfoType | undefined>;
+  replace?: (detectedEntity: SensitiveInfoType) -> string | undefined;
+};
+```
+
+- `chatModel` or `llm`: The chat model or llm that you are wrapping. (eg:
+  `OpenAIChat`)
+- `redact`: The list of entities that you wish to redact. If undefined then all
+  entities are redacted. Valid values are: `email`, `phone-number`,
+  `ip-address`, '`credit-card`, or any string returned from `detect`.
+- `contextWindowSize` - How many tokens to pass to the `detect` function at a
```

Comment:
Can you explain why you wouldn't just set this to a huge number to pass in the full context?

Reply:
Will add a sentence about it. The larger it is the more memory that we need to allocate to copy strings from wasm back in to js to run the function, if they have a large piece of text that they're redacting then it would double the memory footprint of using redact.